javascript/features/death_match/death_match_commands.js

```diff
@@ -47,8 +48,15 @@ export class DeathMatchCommands {
     async enterDeathMatchZone(zone, player) {
         let decision = null;
 
+    const canEnterDeathmatchZone = this.limits().canEnterDeathmatchZone(player);
+
         // If the |player| is switching to a new deathmatch zone, we allow them to switch when they
-        // haven't been shot recently. Otherwise we need to meet the full minigame criteria.
+        // haven't been shot recently or if they aren't in vehicle. Otherwise we need to meet the full minigame criteria.
+        if (canEnterDeathmatchZone.isApproved()) {
```

Comment:
If it's approved I believe you _don't_ want to be able to enter the DM zone.

Are you sure you tested it?